client/src/components/ReportCalendar.js

```diff
@@ -59,9 +61,33 @@ const ReportCalendar = ({
         info.jsEvent.preventDefault()
       }}
       calendarComponentRef={calendarComponentRef}
+      eventRender={renderEvent}
     />
   )
 
+  function renderEvent(info) {
+    let el
+    if (info.view.type === "listDay") {
+      el = info.el.querySelector(".fc-list-item-title > a")
+    } else {
+      el = info.el.querySelector(".fc-title")
+      if (el) {
+        el.style.display = "inline-flex"
+        el.style.alignItems = "center"
+        el.style.verticalAlign = "bottom"
+      }
+    }
+
+    const rprt = new Report(info.event.extendedProps)
+
+    if (el && rprt && rprt.uuid && rprt.engagementDate) {
+      ReactDOM.render(
+        <PlanningConflictForReport report={rprt} text={info.event.title} />,
```

Comment:
if we have a calendar with 100 events in it, will that result in 100 gql querries to find out conflicts?

Reply:
That is correct. In `PlanningConflictForReport` component we use the following query for querying `attendedReports` of `attendees` of a report.
```
  query($uuid: String, $attendedReportsQuery: ReportSearchQueryInput) {
    report(uuid: $uuid) {
      uuid
      engagementDate
      duration
      attendees {
        uuid
        name
        attendedReports(query: $attendedReportsQuery) {
          list {
            uuid
            engagementDate
            duration
          }
        }
      }
    }
  }
```
`$attendedReportsQuery` consists of following two variables `engagementDateStart` and `engagementDateEnd`.

With this approach, we need to know a reports engagement date and duration before querying its attendees with their attended reports. And this is how `PlanningConflictForReport` works.

I considered using an empty `$attendedReportsQuery` query and fetch all attendedReports of all attendees in a single request and do the filtering in client. But that seemed less feasible to me because data size for such a query may increase exponentially over time. With our current approach, we can at least expect a linear number of queries. i.e. at worst we query reports for 42 days which are authored by same person (Actually, I am wondering how many reports one can write in 42 days?).

As a matter of fact; this problem was the reason why we decided to implement this feature with [these graphql changes](https://github.com/NCI-Agency/anet/blob/3b2882e5e1667f46f726e63134099997c6b09222/client/src/components/ReportCalendar.js#L25) as first place.

I couldn't think of a better way of doing this with one or two queries without changing graphql schema or without fetching all attended reports for every attendee.

Reply:
Conflicts are not shown in `dayGridMonth` and `timeGridWeek` views any more.